src/remoted/secure.c

```diff
@@ -944,87 +919,83 @@ STATIC void HandleSecureMessage(const message_t *message, w_indexed_queue_t * co
     os_free(agent_ip);
     os_free(agent_name);
 }
 
-void router_message_forward(char* msg, const char* agent_id, const char* agent_ip, const char* agent_name) {
-    // Both syscollector delta and sync messages are sent to the router
-    ROUTER_PROVIDER_HANDLE router_handle = NULL;
-    int message_header_size = 0;
-    int schema_type = -1;
+void router_message_forward(char* msg, size_t msg_length, const char* agent_id, const char* agent_ip, const char* agent_name) {
 
-    if(strncmp(msg, SYSCOLLECTOR_HEADER, SYSCOLLECTOR_HEADER_SIZE) == 0) {
-        if (!router_syscollector_handle) {
-            mdebug2("Router handle for 'syscollector' not available.");
-            return;
-        }
-        router_handle = router_syscollector_handle;
-        message_header_size = SYSCOLLECTOR_HEADER_SIZE;
-        schema_type = MT_SYS_DELTAS;
-    } else if(strncmp(msg, DBSYNC_HEADER, DBSYNC_HEADER_SIZE) == 0) {
-        if (!router_rsync_handle) {
-            mdebug2("Router handle for 'rsync' not available.");
-            return;
-        }
+    mdebug2("Forwarding message to router: %s", msg);
 
-        int message_subheader_size = 0;
+    // Check if we have the router handle available
+    if (!router_sync_handle) {
+        mdebug2("Router handle for 'inventory synchronization' not available.");
+        return;
+    }
 
-        if(strncmp(msg+DBSYNC_HEADER_SIZE, SYSCHECK_FILE_HEADER, SYSCHECK_FILE_HEADER_SIZE) == 0) {
-            message_subheader_size = SYSCHECK_FILE_HEADER_SIZE;
-        }
-        else if(strncmp(msg+DBSYNC_HEADER_SIZE, SYSCHECK_REGISTRY_KEY_HEADER, SYSCHECK_REGISTRY_KEY_HEADER_SIZE) == 0) {
-            message_subheader_size = SYSCHECK_REGISTRY_KEY_HEADER_SIZE;
-        }
-        else if(strncmp(msg+DBSYNC_HEADER_SIZE, SYSCHECK_REGISTRY_VALUE_HEADER, SYSCHECK_REGISTRY_VALUE_HEADER_SIZE) == 0) {
-            message_subheader_size = SYSCHECK_REGISTRY_VALUE_HEADER_SIZE;
-        }
-        else if (strncmp(msg+DBSYNC_HEADER_SIZE, SYSCOLLECTOR_SYNC_HEADER, SYSCOLLECTOR_SYNC_HEADER_SIZE) == 0) {
-            message_subheader_size = SYSCOLLECTOR_SYNC_HEADER_SIZE;
-        }
-        else {
-            mdebug2("Syscheck message not recognized %s", msg);
-            return;
-        }
+    // Check if message starts with inventory sync header
+    if (strncmp(msg, INVENTORY_SYNC_HEADER, INVENTORY_SYNC_HEADER_SIZE) != 0) {
+        mdebug2("Message does not have inventory sync header.");
+        return;
+    }
 
-        router_handle = router_rsync_handle;
-        message_header_size = DBSYNC_HEADER_SIZE + message_subheader_size;
-        schema_type = MT_SYNC;
-    } else if(strncmp(msg, SYSCHECK_HEADER, SYSCHECK_HEADER_SIZE) == 0) {
-        if (!router_syscheck_handle) {
-            mdebug2("Router handle for 'syscheck' not available.");
-            return;
-        }
+    // Validate minimum message length: header + "x:y" (4 chars minimum after header)
+    if (msg_length <= INVENTORY_SYNC_HEADER_SIZE + 4) {
```

Comment:
Use a constant instead of a magic number:
```suggestion
    #define MIN_MESSAGE_AFTER_HEADER 4
    if (msg_length <= INVENTORY_SYNC_HEADER_SIZE + MIN_MESSAGE_AFTER_HEADER) {
```

Reply:
Done!